src/maintainer/infrastructure.rst

```diff
@@ -266,3 +266,37 @@ To skip a CI build for a given commit, put ``[ci skip] ***NO_CI***`` in the comm
 
   - **abort builds with [skip ci]/etc** `(conda-forge.github.io/#629) <https://github.com/conda-forge/conda-forge.github.io/issues/629>`__
   - **Skip CI requests** `(staged-recipes/#1148) <https://github.com/conda-forge/staged-recipes/issues/1148>`__
+
+
+Compilers and Runtimes
+======================
+
+Conda-forge builds and maintains its own set of compilers for various languages
+and/or systems (e.g., ``C``, ``FORTRAN``, ``C++``, ``CUDA``, etc.). These are used
+in all of our CI builds to build both core dependencies (e.g., ``Python``) and maintainer-contributed
+packages. While we do not have any formal policies or promises of support for these
+compilers, we have historically maintained them according to the following (non-binding)
+principles.
+
+* The authoritative source of the current compilers and versions for various languages
+  and platforms is the `conda_build_config.yaml <https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml>`_
+  in the `conda-forge/conda-forge-pinning-feedstock <https://github.com/conda-forge/conda-forge-pinning-feedstock>`_
+  as described in :ref:`globally_pinned_packages`.
+* We provide no support of any kind in terms of the long-term stability of these pinnings.
+* We upgrade them in an ad-hoc manner on a periodic basis as we have the time and energy to do so.
+  Note that because of the way we enforce runtime constraints, these compiler upgrades will not break
+  existing packages. However, if you are using the compilers outside of ``conda``, then you may find issues.
+* We generally provide notice in the form of an announcement when a compiler is upgraded.
+  Note that these changes take months to complete, so you will generally have a lot of time
```

Comment:
may take months? 
I do not think the the recent gfortran or clang migrations on macOS took that long and the VS 2015 -> 2017 migration should be very expedient?

Reply:
Happy to be less specific here. The move to gcc9 will probably be this big but others are smaller yes.

Reply:
done. LMK!

Reply:
> The move to gcc9 will probably be this big but others are smaller yes.

Not really. gcc-4->gcc-7 was brutal because, 1. C++ ABI changed, 2. we switched to a different compiler. `gcc-7` -> `gcc-9` would be smooth and only requires Fortran packages to be updated.

Reply:
Right. OK.